torch/_inductor/codegen/cpp_micro_gemm.py

```diff
@@ -556,6 +615,45 @@ class CppMicroGemmAMX(CppMicroGemm):
         zero_c();
     }
 
+    {%- if input_dtype == torch.bfloat16 and input2_dtype == torch.int8 %}
+    // create a buffer for tiles of B.
+    // To make things simpler, use the stack instead of calling a memory allocator.
+    // allocating an array of size 1024 + 64, and then using an address that's 64 bytes aligned.
+    // TODO: loop-unrolling of the "compute" lambda may result in incorrect output
+    // as this buffer would be used, so maybe 4 of these should be used?
+    // Since UT output is correct, looks like loop unrolling isn't actually happening.
+    {{input_t}} bf16_weights_buf[544];
+    int bf16_weights_buf_ptr_mod = (uintptr_t)((void*)bf16_weights_buf) % 64;
+    {{input_t}}* bf16_weights_buf_ptr =
+        bf16_weights_buf_ptr_mod == 0
+            ? bf16_weights_buf
+            : ({{input_t}}*)((uintptr_t)((void*)bf16_weights_buf) + 64 - bf16_weights_buf_ptr_mod);
+    memset(bf16_weights_buf_ptr, 0, 1024);
+
+    int num_b_rows = (last_k_offset > 0) ? 16 : (tail_k_size * sizeof({{input_t}})) / 4;
+    int b_tile_ptr_stride = ldb * {{vnni_size}};
+
+    // TODO: verify whether or not these lambdas inline
+    auto load_B_row = [&]({{input2_t}}* src, {{input_t}}* dst) {
+        {{kernel.unroll_pragma(2)}}
+        for (int i = 0; i < 2; i++) {
+            auto b32 = at::vec::convert_to_int32<int8_t>(src + i * 16);
+            auto b_bf16 = at::vec::convert<{{input_t}}>(b32);
```

Comment:
Can we load int8 and convert to `input_t` directly? Not sure if it benefits perf.

Reply:
Such an approach would entail first loading int8 values in an `__m256i`, and then using its 2 `__m128i` halves of 16 elements each to convert it to int32 in separate `__m512i`s, followed by conversion to float (in `__m512`), and subsequently 16 BFloat16 each.

So, I think it's not worth pursuing.
Please advise. Thanks!

Reply:
Let's keep the current implementation.